### PR TITLE
Add clip op

### PR DESCRIPTION
**Pull request by kahmed10**

### CODE REVIEW

File: src/targets/cpu/lowering.cpp [hunk 1]
```diff
@@ -140,6 +140,21 @@ struct cpu_lrn
...
 };
 
+struct clip_op
+{
+    op::clip op;
+    std::string name() const { return "cpu::clip"; }
+    auto fcn() const
+    {
+        auto& max = op.max_val;
+        auto& min = op.min_val;
```
Comment (pfultz2) [inline src/targets/cpu/lowering.cpp]: These variables can be values and not references.

File: src/targets/gpu/clip.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+shape hip_clip::compute_shape(std::vector<shape> inputs) const
+{
+    inputs.pop_back();
+    return op.compute_shape(inputs);
+}
+
+argument
+hip_clip::compute(context& ctx, const shape& output_shape, const std::vector<argument>& args) const
+{
+    assert(output_shape == args.back().get_shape());
```
Comment (pfultz2) [inline src/targets/gpu/clip.cpp]: What is the reason for this assert?

Comment (kahmed10) [reply src/targets/gpu/clip.cpp]: This was copied from another compute function, so I don't know if it's necessary.

Comment (pfultz2) [reply src/targets/gpu/clip.cpp]: Well if the allocation is different we now have the `adjust_gpu_allocation` pass that will fix these allocations.

File: src/targets/gpu/include/migraphx/gpu/device/clip.hpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+namespace device {
+
+void clip(hipStream_t stream,
+          const argument& result,
+          const argument& arg1,
+          const float& max,
```
Comment (pfultz2) [inline src/targets/gpu/include/migraphx/gpu/device/clip.hpp]: These floats should be passed by value.

**Pull request merged**